File: p1.cpp
#include <iostream>
#include <vector>
#include <array>

using namespace std;


class head{
public:
  std::vector<int> head;
};

class Graph{
public:
  int v;
  std::vector<std::vector<int> > nodeList;
  Graph(int v);
  void addEdge(int source, int dest);
  std::vector<std::vector<int> > getCopy(){
    return nodeList;
  }
  
};

void Graph::addEdge(int source, int dest){
  nodeList[source].push_back(dest);
  nodeList[dest].push_back(source);
}

Graph :: Graph(int val){
  v = val;
  for(int i=0; i<v;i++){
    nodeList.push_back(vector<int>());
  }
}


void write(const Graph &g){
  while (!g.nodeList.empty()){
    while(!g.nodeList.back().empty()){
      cout<<g.nodeList.back().back();
      g.nodeList.back().pop_back();
    }
    g.nodeList.pop_back();
  }

}
int main(){
  Graph g(4);
  cout<<"test";
  g.addEdge(1,2);
  g.addEdge(1,3);
  write(g);
  return 0;
}

File: p1.py



class head:
    def __init__(self, head=None):
        self.head = []

class Graph:
    def __init__(self,v):
        self.v = v
        self.nodeList = []
        for i in range(v):
            self.nodeList.append(head())
            
    def addEdge(self, source, dest):
        self.nodeList[source].head.append(dest)
        self.nodeList[dest].head.append(source)
        
    def __str__(self):
        output = ''
        output += str(self.v)
        for i in range(len(self.nodeList)):
            output+=str(len(self.nodeList[i].head))
            output+=''.join(map(str,self.nodeList[i].head))
        return output

def read(Graph):
    pass

def write(Graph):
    print(Graph)

def main():
    pass

        
        
